subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java

```diff
@@ -96,6 +94,12 @@
         DID_NO_WORK
     }
 
+    enum Incrementality {
+        NOT_INCREMENTAL,
+        INCREMENTAL,
+        LEGACY_INCREMENTAL
```

Comment:
At least `LEGACY_INCREMENTAL` needs explanation. Why is it "legacy"? What does it do differently to `INCREMENTAL`? How is it going away?

Reply:
I renamed the constants.